src/map/Map.tsx

```diff
@@ -1,51 +1,26 @@
-import { coordinateToText } from '@/Converters'
-import ReactMapGL, { Layer, MapEvent, Marker, Popup, Source } from 'react-map-gl'
+import ReactMapGL, { MapEvent } from 'react-map-gl'
 import 'mapbox-gl/dist/mapbox-gl.css'
-import { Coordinate, QueryPoint } from '@/stores/QueryStore'
-import React, { useState } from 'react'
+import { Coordinate } from '@/stores/QueryStore'
+import React from 'react'
 import Dispatcher from '@/stores/Dispatcher'
-import { MapIsLoaded, SetPoint, SetSelectedPath, SetViewport } from '@/actions/Actions'
-import { Path } from '@/api/graphhopper'
-import { PathDetailsPoint } from '@/stores/PathDetailsStore'
+import { MapIsLoaded, SetViewport } from '@/actions/Actions'
 import { RasterStyle, StyleOption, VectorStyle } from '@/stores/MapOptionsStore'
-import { FeatureCollection, LineString } from 'geojson'
-import { PopupComponent } from '@/map/Popup'
-import { MarkerComponent } from '@/map/Marker'
 import { ViewportStoreState } from '@/stores/ViewportStore'
-
-const pathsLayerKey = 'pathsLayer'
-const selectedPathLayerKey = 'selectedPathLayer'
-const highlightedPathSegmentLayerKey = 'highlightedPathSegmentLayer'
+import { MapLayer } from '@/stores/MapLayerStore'
 
 type MapProps = {
     viewport: ViewportStoreState
-    selectedPath: Path
-    paths: Path[]
-    queryPoints: QueryPoint[]
     mapStyle: StyleOption
-    pathDetailPoint: PathDetailsPoint | null
-    highlightedPathDetailSegments: Coordinate[][]
+    mapLayers: { [key: string]: MapLayer }
+    setPopupCoordinate: (c: Coordinate | null) => void
```

Comment:
Passing around the `setPopupCoordinate` function here is against the current architecture and we probably would need another store for this. So far we were able to keep `const [popupCoordinate, setPopupCoordinate] = useState(null)` in `Map.tsx`, but now that `Map` and `ContextMenuLayer` are separate components this is more complicated.

Reply:
Hm, or we move the `ContextMenuLayer` back into `Map.tsx` because it is kind of more related to the map than the others? :) Btw. the content of the context menu is definitely something that we might need to update based on user-interaction, e.g. when switching between /route and /isochrone mode for example.

Reply:
... this is what I did :)